multiqc/modules/supernova/supernova.py

```diff
@@ -0,0 +1,412 @@
+                'smooth_points_sumcounts': False,
+                'xmax': 200 # Emulate the PDF output of supernova
+            }
+            self.add_section (
+                name = 'K-mer counts',
+                anchor = 'supernova-kmers',
+                description = 'Shows the k-mer frequencies of the input data to Supernova (after filtering)',
+                helptext = 'This data is generated from k-merizing the input read data, where the sequences are ' \
+                        'transformed in to the set of all possible sub-sequences of a fixed length of K (Supernova uses K=48). ' \
+                        'The plot shows on the x-axis the multiplicity (i.e. how many times are they repeated) of these k-mers ' \
+                        'and the y-axis the number of k-mers at this level of multiplicity. ' \
+                        'A careful reading of this plot can give some insights into the levels of heterozygosity and repeats ' \
+                        'in the genome that was sequenced and indications if the sequencing experiment was successful.',
+                plot = linegraph.plot(kmers, config_kmers)
+            )
+
+
+    def parse_summary(self, content):
+
+        stats = {
+            'assembly_size': 'Asm size',
+            'bases_per_read': 'Read len',
+            'contig_N50': 'Contig N50',
+            'dup_perc': '% Dup',
+            'edge_N50': 'Edge N50',
+            'effective_coverage': 'Coverage',
+            'hetdist': 'Het dist',
+            'lw_mean_mol_len': 'Mol size',
+            'median_ins_sz': 'Insert size',
+            'nreads': '# Reads',
+            'phase_block_N50': 'Phase N50',
+            'placed_perc': '% Phased',
+            'placed_frac': '% Phased',
+            'proper_pairs_perc': '% proper',
+            'q30_r2_perc': '% R2 Q30',
+            'rpb_N50': 'Barcode N50',
+            'scaffold_N50': 'Scaff N50',
+            'scaffolds_10kb_plus': '# Long scaffs',
+            'valid_bc_perc': '% missing BC'
+        }
+ 
+        try:
+            cdict = json.loads(content)
+        except ValueError as e:
+            raise e
+
+        data = {}
+        # Try to find sample_id
+        sid = ''
+        if 'CS_SAMPLE_ID' in cdict.keys():
+            sid = cdict['CS_SAMPLE_ID'] # supernova 1.2
+        elif 'sample_id' in cdict.keys():
+            sid = cdict['sample_id']
+        else:
+            raise RuntimeError
+
+        for key, value in cdict.items():
+            if key in stats.keys():
+                # Some trickery for supernova 1.1.4 compatability
+                if key == 'placed_frac':
+                    value = value * 100
+                if key == 'valid_bc_perc':
+                    value = 100 - value
+                data[stats[key]] = value
+        
+        return (sid, data)
+
+
+    def parse_report(self, content):
+        # Some short-hands for converting the report numbers (pi is exactly three!)
+        exp = {
```

Comment:
I feel as though this a crime against science, but I couldn't think of any other way to keep the `multiqc_data/multiqc_supernova.txt` consistent. Thoughts @ewels ?

Reply:
What's wrong with this approach? That we're making it look like the numbers are more accurate than they really are? I think it's fine... Better to have useful numbers for downstream use than less helpful string values (where people will just do this themselves anyway).

You can always confess to your crime in the docs if you really want ;)

Reply:
Yeah, I guess I'm just worrying too much. And yes, having something useful and not in need of interpretation / conversion of the raw data is what I was going for.

I already confessed my crime with that code cheeky comment :wink: